### ISSUE TITLE
Proposal: Online Chess game integrated with Kaia blockchain. 2 people compete with each other. Whoever loses loses 1 Klay. Whoever wins gains 1 Kaly.

### ISSUE THREAD

**Issue by linHDev0106** [kaiachain/kaia-dapp-mono#145]
Description
- The project is a 2-player online chess game.
- Players connect to Kaia via Metamask wallet.
- Before playing, each player bets 1 Klay.
- 2 players play chess with each other. Whoever loses will lose 1 Klay. Whoever wins will get 1 Klay.

Bounty track
- Other

![Screenshot 2024-09-28 162726](https://github.com/user-attachments/assets/34fe0384-1f01-4ad5-8b52-12db4bdcf6f0)

**Comment by zxstim** [comment]
Is the chess engine onchain?

if not, I would prefer that you work with Chess.com API (https://www.chess.com/news/view/published-data-api) to build an Oracle service

Players would play chess on there and make bets. Whoever wins gets the pot. At the same time, other viewers can bet as well.

**Comment by zxstim** [comment]
Amended the title

**Comment by linHDev0106** [comment]
The game you sent is really good and professional and the algorithm is very complicated.
The game I created is just a 2-player version.
In which 2 people connect to the same room. The first person bets 1 Klay. Before the second person enters the room, if the first person decides to leave, they will receive 1 Klay.
When the second player enters the room, both of them have already bet 1 Klay.
Then the 2 people play together, the winner will receive 2 Klay (1 Klay bet and 1 Klay win).
The game has 3D graphics, not too complicated (because the team has only me, coding the game and coding the blockchain).
Thanks.

**Comment by zxstim** [comment]
I agree. That's the reason why I recommend you to use Chess.com API. You would be primarily building the betting part rather than building the chess game.

Basically, you run a oracle service that watches the chess.com matches. Say 2 people entering chess.com, they can enter chess.com, to start a match and then make a bet on your DApp

This way, the best part is the betting will be completed and polished to great extent first (this is the part that makes you money). Then you can work on the chess game after.
